pkg/web/initclient/init_http_handler_test.go

```diff
@@ -0,0 +1,125 @@
+package initclient
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+
+	"github.com/aerogear/mobile-security-service/pkg/models"
+
+	"github.com/aerogear/mobile-security-service/pkg/web/apps"
+	"github.com/labstack/echo"
+)
+
+func TestHTTPHandler_InitClientApp(t *testing.T) {
+	type fields struct {
+		appsService apps.Service
+	}
+	type args struct {
+		device models.Device
+	}
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantStatusCode int
+	}{
+		{
+			name: "A 400 Bad Request should be returned when request body is missing device ID",
+			args: args{
+				device: models.Device{
+					ID:            "d1895cc1-28d7-4283-932d-8bcab9e4a469",
+					VersionID:     "1a3fbe59-e69c-42c7-b9c0-d7eb7e65c073",
+					AppID:         "com.aerogear.testapp",
+					DeviceVersion: "1.1",
+					Version:       "9.1.1",
+					DeviceType:    "Android",
+				},
+			},
+			wantStatusCode: 400,
+		},
+		{
+			name: "A 400 Bad Request should be returned when request body is missing version",
+			args: args{
+				device: models.Device{
```

Comment:
As described above use the fixture to mock the struct.

Reply:
Using the mock fixture is not suitable here. This is simulating many different scenarios. As you can see the first test case has a device with a missing DeviceID property. Each of the other test cases do something similar expect with slightly different property changes. 

The fixtures only return valid mock data. We could create a new fixture method for each test case but then we are still writing the same amount of code except in a different file.

Reply:
Could you not do something as in order to make clear and easier: 

```go
deviceInvalidId = helpers.GetMockDevice()
deviceInvalidId.deviceId = "invalid" 

```

Reply:
How do you suggest this can be done in the test table format?

```go
{
		{
			name: "A 400 Bad Request should be returned when request body is missing device ID",
			args: args{
				device: models.Device{
					ID:            "d1895cc1-28d7-4283-932d-8bcab9e4a469",
					VersionID:     "1a3fbe59-e69c-42c7-b9c0-d7eb7e65c073",
					AppID:         "com.aerogear.testapp",
					DeviceVersion: "1.1",
					Version:       "9.1.1",
					DeviceType:    "Android",
				},
			},
			wantStatusCode: 400,
		},
		{
			name: "A 400 Bad Request should be returned when request body is missing version",
			args: args{
				device: models.Device{
					ID:            "d1895cc1-28d7-4283-932d-8bcab9e4a469",
					DeviceID:      "1a3fbe59-e69c-42c7-b9c0-d7eb7e65c073",
					AppID:         "com.aerogear.testapp",
					DeviceVersion: "1.1",
					DeviceType:    "Android",
				},
			},
			wantStatusCode: 400,
		},
		{
			name: "A 400 Bad Request should be returned when request body is missing app ID",
			args: args{
				device: models.Device{
					ID:            "d1895cc1-28d7-4283-932d-8bcab9e4a469",
					VersionID:     "1a3fbe59-e69c-42c7-b9c0-d7eb7e65c073",
					DeviceVersion: "1.1",
					Version:       "9.1.1",
					DeviceType:    "Android",
				},
			},
			wantStatusCode: 400,
		},
		{
			name: "Expect init data to be returned when valid device is supplied",
			args: args{
				device: models.Device{
					ID:            "d1895cc1-28d7-4283-932d-8bcab9e4a469",
					VersionID:     "1a3fbe59-e69c-42c7-b9c0-d7eb7e65c073",
					DeviceID:      "db511711-95e8-4da2-8e76-1700465ae8ca",
					AppID:         "com.aerogear.testapp",
					DeviceVersion: "1.1",
					Version:       "9.1.1",
					DeviceType:    "Android",
				},
			},
			wantStatusCode: 200,
		},
```

Reply:
You will put it one top as global value for the func and pass a device as arg.

Reply:
HI @craicoverflow it could be done as [here](https://github.com/aerogear/mobile-security-service/pull/90/files#diff-b910f1425e179a0e8b94ac700b838479R382)